week_10/박서준/Project4/C_거짓말.cpp

```diff
@@ -0,0 +1,85 @@
+﻿#include <iostream>
+#include <vector>
+#include <algorithm>
+using namespace std;
+#define TrueMan 0
+
+int findRoot(int idx, vector<int>& unions)
+{
+	if (unions[idx] == idx)
+		return idx;
+	return unions[idx] = findRoot(unions[idx], unions);
+}
+
+void unionFind(int p1, int p2, vector<int>& unions)
+{
+	int iRoot = findRoot(p1, unions);
+	int jRoot = findRoot(p2, unions);
+
+	if (iRoot != jRoot)
+	{
+		// 값이 낮은 루트로 선택
+		if (iRoot < jRoot)
+		{
+			unions[jRoot] = iRoot;
+		}
+		else
+		{
+			unions[iRoot] = jRoot;
+		}
+	}
+}
+
+int main()
+{
+	ios_base::sync_with_stdio(false);
+	cin.tie(NULL);
+
+	int N, M;
+	cin >> N >> M;
+	vector<int> unions(N + 1);
+	for (int i = 1; i <= N; i++) {
+		unions[i] = i;  // 초기화: 사람 자기 자신이 루트
+	}
+	int trueMans;
+	cin >> trueMans;
+	for (int i = 1;i <= trueMans;i++)
+	{
+		int trueMan;
+		cin >> trueMan;
+		unions[trueMan] = TrueMan;
+	}
```

Comment:
```suggestion
	for (int i = 0;i < trueMans;i++)
	{
		int trueMan;
		cin >> trueMan;
		unions[trueMan] = TrueMan;
	}
```
코드를 짧게 작성함으로써 0.01초의 이득을...!
근데 헷갈릴거면 이렇게 적는게 좋습니다